pkg/validate/validate.go

```diff
@@ -372,3 +375,26 @@ func osSecretFromProfile(ctx context.Context, pType objectstore.ProviderType, p
 	}
 	return secret, nil
 }
+
+func ValidateLabels(labels map[string]string) error {
+	for k, v := range labels {
+		if errs := validation.IsQualifiedName(k); len(errs) > 0 {
+			return errkit.New(fmt.Sprintf("label key '%s' failed validation. %s", k, errs))
+		}
+
+		if errs := validation.IsValidLabelValue(v); len(errs) > 0 {
+			return errkit.New(fmt.Sprintf("label value '%s' failed validation. %s", v, errs))
+		}
```

Comment:
optional: I could see some value in processing the full set and `errors.Join`ing all the problems at the end - so if you happen to have multiple validation failures, you don't resolve them one by one only to discover the next bad one on a separate run...

Reply:
Yep, sounds reasonable, better stay consistent.